validators.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package fed
+
+import (
+	"errors"
+	"regexp"
+)
+
+var (
+	numericRegex = regexp.MustCompile(`[^0-9]`)
+	msgNumeric   = "is not 0-9"
+)
+
+// validator is common validation and formatting of golang types to fed type strings
+type validator struct{}
+
+// isNumeric checks if a string only contains ASCII numeric (0-9) characters
+func (v *validator) isNumeric(s string) error {
```

Comment:
A regex might be slow compared to checking each rune:

```Go
for i := range s {
	if s[i] < '0' || s[i] > '9' {
		return errors.New(msgNumeric)
	}
}
```

Reply:
I believe I tested that many moons ago and it was negligible